converter/converter/v1_v2_converter.py

```diff
@@ -0,0 +1,168 @@
+        "FEM": "F",
+        "AUTRE": "O",
+        "INCONNU": "UN"
+    }
+
+    WHATS_HAPPEN_CODE_MAPPING = {
+        "C02.15.00": "C02.16.00",
+        "C02.15.01": "C02.16.01",
+        "C02.15.02": "C02.16.02",
+        "C02.15.03": "C02.16.03"
+    }
+
+    DETAIL_ATTRIBUTION_MAPPING ={
+        "DRM":"DRM",
+        "DRMMRU":"DRM.MRU",
+        "MU":"DRM.MRU.MU",
+        "DENT":"DRM.SPE.DENT",
+        "GERIA":"DRM.SPE.GERIA",
+        "PEDIA":"DRM.SPE.PEDIA",
+        "PSY":"DRM.SPE.PSY",
+        "TOXICOL":"DRM.SPE.TOXICOL",
+        "INDISPMU:":"DRM.MRU.INDISPMU",
+        "SSE":"DRM.MRU.SSE",
+        "PLANBLAN":"DRM.MRU.PLANBLAN",
+        "PCSAMU:":"DRM.MRU.PCSAMU",
+        "DRMMRL":"DRM.MRL",
+        "MG":"DRM.MRL.MG",
+        "INDISPMG":"DRM.MRL.INDISPMG",
+        "ABSML":"DRM.MRL.ABSML",
+        "DR":"DR",
+        "DREG":"DR.DREG",
+        "DRARM":"DR.DREG.DRARM",
+        "DRMED":"DR.DREG.DRMED",
+        "DRPHARMA":"DR.DREG.DRPHARMA",
+        "DRDENT":"DR.DREG.DRDENT",
+        "DRINFO":"DR.DREG.DRINFO",
+        "DOS-SIS":"DR.DREG.DOS-SIS",
+        "DOS-FDO":"DR.DREG.DOS-FDO",
+        "D":"D",
+        "D-MALV":"D.D-MALV",
+        "ERR":"D.D-MALV.ERR",
+        "NRP":"D.D-MALV.NRP",
+        "MALV":"D.D-MALV.MALV",
+        "FAX":"D.D-MALV.FAX",
+        "ITERATIF":"D.D-MALV.ITERATIF",
+        "D-IDENT":"D.D-IDENT",
+        "ADMIN":"D.D-IDENT.ADMIN",
+        "PERSO":"D.D-IDENT.PERSO",
+        "AUTRE":"D.D-IDENT.AUTRE",
+    }
+
+    CALLER_TYPE_MAPPING = {
+        "MEDSOS":"MED.MEDSOS",
+        "MRL":"MED.MRL",
+        "EFFML":"MED.EFFML",
+        "INF":"SANTE.INF",
+        "AIDESOIN":"SANTE.AIDESOIN",
+        "SF":"SANTE.SF",
+        "AIDEDOM":"SANTE.AIDEDOM",
+    }
+
+    DECISION_RESOURCE_TYPE_MAPPING = {
+        "LIB":"LIBERAL",
+    }
+
+    @classmethod
+    def upgrade(cls, input_json: Dict[str, Any]) -> Dict[str, Any]:
+        def validate_diagnosis_code(json_data:Dict[str, Any],diagnosis_type:str):
+            code = get_field_value(json_data, f"$.hypothesis.{diagnosis_type}.code")
+            if code != None:
+                pattern = re.compile(cls.DIAGNOSIS_CODE_VALIDATION_REGEX)
+                correct_format = pattern.match(code)
+
+                if not correct_format:
+                    delete_paths(json_data, [f"hypothesis.{diagnosis_type}"])
+
+        # todo : move to utils and reuse it where needed (+ add test)
+        def update_json_value(data, jsonpath_query, new_value):
+            jsonpath_expr = parse(jsonpath_query)
+            matches = jsonpath_expr.find(data)
+            matches[0].full_path.update(data, new_value)
+
+        def map_to_new_value(json_data: Dict[str,Any], json_path: str, mapping_value : Dict[str,str]):
+            current_value = get_field_value(json_data, json_path)
+
+            if current_value != None:
+                new_value = mapping_value.get(current_value, current_value)
+
+                if new_value != current_value:
```

Comment:
Les goûts et les couleurs à nouveau mais que penses-tu de :
```suggestion
            if current_value != None and current_value in mapping_value:
                new_value = mapping_value.get(current_value)
```